### ISSUE TITLE
github build failure

### ISSUE THREAD

**Issue by losemyheaven** [hexojs/hexo#3138]
## Environment Info

Node version(`node -v`): node -v

Your site `_config.yml` (Optional): https://gist.github.com/losemyheaven/53866b8c69bdbcb6149ef501289b5cec

Your theme `_config.yml` (Optional): 
https://gist.github.com/losemyheaven/53866b8c69bdbcb6149ef501289b5cec (bottom half)

Hexo and Plugin version(`npm ls --depth 0`): 
hexo-theme-next@5.1.4 /home/lost/code/losemyheaven.github.io/themes/next
└── (empty)

## For BUG
1. BUG description
github note me build failure due to HISTORY.md.
But `hexo s` works fine and I can get the right page.

> The page build failed for the `master` branch with the following error:
> 
> There is a syntax error in the 'for' loop on line 118 of `node_modules/hexo-deployer-git/node_modules/swig/HISTORY.md`. Valid syntax is 'for <item> in <collection>', and the <item> variable may not contain hyphens or quotation marks. For more information, see https://help.github.com/articles/page-build-failed-syntax-error-in-for-loop/.
> 
> For information on troubleshooting Jekyll see:
> 
>   https://help.github.com/articles/troubleshooting-jekyll-builds
> 
> If you have any questions you can contact us by replying to this email.

2. The way to reproduce
you can download my github.io repo
https://github.com/losemyheaven/losemyheaven.github.io.git
there is two branch, hexo branch is for hexo

**Comment by yujianyuanhaha** [comment]
I also met exactly same problem.

**Comment by yujianyuanhaha** [comment]
I just start from scratch over from 'blog init', and fix the problem.

**Comment by fitsyu** [comment]
Hi,
I was suffering from this issue as well.
I followed deployment guide from the [doc.](http://hexo.io/docs/deployment)

Today I realize that Github Pages could only be generated from either master or gh-pages branch as we can see in repository settings.
So I changed branch from `published`to `master`in deploy section in _config.yml file.

However this caused my regular workflow's change.
I have to use different branch to work and make `master`branch as publicaction branch.

Hope it helps.
